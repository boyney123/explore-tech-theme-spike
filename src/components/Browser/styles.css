.browser {
  margin-top: 20px;
}

.browser__body img {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  border: 1px solid #bcbcbc;
  border-radius: 0 0 4px 4px;
  box-shadow: 0 0 8px transparent;
  height: auto;
  transition: box-shadow 0.2s ease;
  width: 100%;
  padding: 2px;
}

.browser__body img:hover {
  box-shadow: 0 8px 24px rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.browser__header {
  padding: 10px;
  background: #f1f1f1;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.browser__header:after {
  content: '';
  display: table;
  clear: both;
}

.browser__header__dots {
  float: left;
}

.browser__header__dots__dot {
  margin: 4px 4px 0 0;
  height: 12px;
  width: 12px;
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
}

.browser__header__dots__dot:nth-child(1) {
  background: #ed594a;
}

.browser__header__dots__dot:nth-child(2) {
  background: #fdd800;
}

.browser__header__dots__dot:nth-child(3) {
  background: #5ac05a;
}

.browser .address-bar {
  width: 75%;
  float: left;
}

.browser .address-bar input {
  width: 100%;
  border-radius: 3px;
  border: none;
  background-color: white;
  margin-top: -8px;
  height: 25px;
  color: #666;
  padding: 5px;
  font-size: 0.7em;
}

.browser__footer {
  text-align: center;
}

.browser__footer h4 {
  margin: 0;
  padding: 0;
}
