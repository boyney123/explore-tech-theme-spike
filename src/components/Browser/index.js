import React from 'react'
import PropTypes from 'prop-types'
import { navigate } from 'gatsby'
import Img from 'gatsby-image'

import './styles.css'

const Browser = ({
  path,
  title,
  author,
  subtitle,
  github_url,
  image = {},
} = {}) => {
  const { childImageSharp: { fluid: screenshot } = {} } = image

  return (
    <div className="column is-one-third">
      <div className="browser" onClick={() => navigate(path)}>
        <div className="browser__header">
          <div className="browser__header__dots">
            <span className="browser__header__dots__dot" />
            <span className="browser__header__dots__dot" />
            <span className="browser__header__dots__dot" />
          </div>
        </div>
        <div className="browser__body">
          <Img fluid={screenshot} />
        </div>
        <div className="browser__footer">
          <h4 className="mp0">{title}</h4>
          <p> by {author}</p>
        </div>
      </div>
    </div>
  )
}

Browser.propTypes = {
  path: PropTypes.string,
  title: PropTypes.string,
  author: PropTypes.string,
  subtitle: PropTypes.string,
  github_url: PropTypes.string,
  image: PropTypes.object,
}

export default Browser
