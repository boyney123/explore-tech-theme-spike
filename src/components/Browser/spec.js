import React from 'react'
import { shallow } from 'enzyme'
import { navigate } from 'gatsby'

import Browser from './'

describe('Browser', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Browser />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with the given gatsby childImageSharp fluid image', () => {
    const image = {
      childImageSharp: {
        fluid: { aspectRatio: 10, src: 'test.png', srcSet: '', sizes: '1' },
      },
    }

    const wrapper = shallow(<Browser image={image} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows the given title and author under the browser', () => {
    const wrapper = shallow(<Browser title="My new component" author="David" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('when clicking on the browser user is navigated to the components `path` prop', () => {
    const wrapper = shallow(<Browser path="/some-random-path" />)
    wrapper.find('.browser').simulate('click')
    expect(navigate).toHaveBeenCalledWith('/some-random-path')
  })
})
