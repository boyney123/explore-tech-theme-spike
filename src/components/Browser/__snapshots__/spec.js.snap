// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser renders correctly 1`] = `
<div
  className="column is-one-third"
>
  <div
    className="browser"
    onClick={[Function]}
  >
    <div
      className="browser__header"
    >
      <div
        className="browser__header__dots"
      >
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
      </div>
    </div>
    <div
      className="browser__body"
    >
      <Image
        Tag="div"
        alt=""
        critical={false}
        fadeIn={true}
      />
    </div>
    <div
      className="browser__footer"
    >
      <h4
        className="mp0"
      />
      <p>
         by 
      </p>
    </div>
  </div>
</div>
`;

exports[`Browser renders with the given gatsby childImageSharp fluid image 1`] = `
<div
  className="column is-one-third"
>
  <div
    className="browser"
    onClick={[Function]}
  >
    <div
      className="browser__header"
    >
      <div
        className="browser__header__dots"
      >
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
      </div>
    </div>
    <div
      className="browser__body"
    >
      <Image
        Tag="div"
        alt=""
        critical={false}
        fadeIn={true}
        fluid={
          Object {
            "aspectRatio": 10,
            "sizes": "1",
            "src": "test.png",
            "srcSet": "",
          }
        }
      />
    </div>
    <div
      className="browser__footer"
    >
      <h4
        className="mp0"
      />
      <p>
         by 
      </p>
    </div>
  </div>
</div>
`;

exports[`Browser shows the given title and author under the browser 1`] = `
<div
  className="column is-one-third"
>
  <div
    className="browser"
    onClick={[Function]}
  >
    <div
      className="browser__header"
    >
      <div
        className="browser__header__dots"
      >
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
        <span
          className="browser__header__dots__dot"
        />
      </div>
    </div>
    <div
      className="browser__body"
    >
      <Image
        Tag="div"
        alt=""
        critical={false}
        fadeIn={true}
      />
    </div>
    <div
      className="browser__footer"
    >
      <h4
        className="mp0"
      >
        My new component
      </h4>
      <p>
         by 
        David
      </p>
    </div>
  </div>
</div>
`;
